File: scripts/vega/example_1.py
import altair as alt
import pandas as pd

df = pd.read_parquet('sales.parquet')

# Interactive linked charts
brush = alt.selection_interval()

points = alt.Chart(df).mark_point().encode(
    x='revenue:Q',
    y='profit:Q',
    color=alt.condition(brush, 'category:N', alt.value('lightgray'))
).add_selection(brush).properties(width=400, height=300)

bars = alt.Chart(df).mark_bar().encode(
    x='count()',
    y='category:N',
    color='category:N'
).transform_filter(brush).properties(width=400, height=300)

chart = points | bars
chart.save('dashboard.html')

File: scripts/vega/example_1.js
// Interactive scatter plot with Vega-Lite
const spec = {
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "data": { "url": "/api/sales" },
  "mark": "point",
  "encoding": {
    "x": {
      "field": "revenue",
      "type": "quantitative",
      "scale": { "zero": false }
    },
    "y": {
      "field": "profit",
      "type": "quantitative"
    },
    "color": {
      "field": "category",
      "type": "nominal"
    },
    "size": {
      "field": "volume",
      "type": "quantitative"
    },
    "tooltip": [
      { "field": "product", "type": "nominal" },
      { "field": "revenue", "type": "quantitative", "format": "$,.0f" }
    ]
  },
  "selection": {
    "brush": { "type": "interval" }
  },
  "width": 600,
  "height": 400
};

// Embed in page
vegaEmbed('#chart', spec, { theme: 'dark' });